pkgs/by-name/pa/parmetis/package.nix

```diff
@@ -0,0 +1,38 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  metis,
+  mpi,
+}:
+
+stdenv.mkDerivation {
+  pname = "parmetis";
+  version = "4.0.3";
+
+  src = fetchFromGitHub {
+    owner = "KarypisLab";
+    repo = "ParMETIS";
+    rev = "d90a2a6cf08d1d35422e060daa28718376213659";
+    hash = "sha256-22YQxwC0phdMLX660wokRgmAif/9tRbUmQWwNMZ//7M=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  enableParallelBuilding = true;
```

Comment:
```suggestion

```
already done by cmake

Reply:
Are you sure?
If i remove the line it seems like the build takes more time:

```
# command used: `time nix-build . -A parmetis`
# with `enableParallelBuilding = true;`
nix-build . -A parmetis  1.68s user 0.34s system 8% cpu 22.698 total
# vs without `enableParallelBuilding = true;`
nix-build . -A parmetis  1.67s user 0.35s system 3% cpu 57.441 total
```

Reply:
You probably want `[ cmake ninja ]`.

Reply:
I think you mentioned this before: 
![image](https://github.com/user-attachments/assets/5a1b8c5e-ef3e-4ebe-baae-6950f54b3509)

https://github.com/NixOS/nixpkgs/pull/329102#discussion_r1689175243

Or am i using this incorrectly? I just added it to the nativeBuildInputs

Reply:
Uh, sorry, I completely forgot, I was just reacting to the parallel build stuff… it *should* just work with CMake builds, but I guess there’s something funky going on here. I think you’d have to drop the `make` stuff in `configurePhase` and just let the CMake hook run naturally, with any required `cmakeFlags` corresponding to [the ones set in the upstream `Makefile`](https://github.com/KarypisLab/ParMETIS/blob/8ee6a372ca703836f593e3c450ca903f04be14df/Makefile).

Reply:
I tested several different approaches: Running the `tar xf` during preConfigure and not touching the configurePhase, commenting out `make` in the configurePhase, but nothing worked. How bad would it be to just set `enableParallelBuilding = true;`? I can also remove the line and we can settle for only building with one thread if the option is that bad...

Reply:
When setting the cmakeFlags to 
```
cmakeFlags = [
  "-DGKLIB_PATH=metis/GKlib"
  "-DMETIS_PATH=metis" 
];
```

The build is running in parallel as expected, but then the build fails with
```
parmetis> [ 55%] Building C object libparmetis/CMakeFiles/parmetis.dir/balancemylink.c.o
parmetis> [ 55%] Building C object libparmetis/CMakeFiles/parmetis.dir/akwayfm.c.o
parmetis> [ 56%] Building C object libparmetis/CMakeFiles/parmetis.dir/comm.c.o
parmetis> In file included from /build/source/libparmetis/balancemylink.c:14:
parmetis> /build/source/libparmetis/./parmetislib.h:23:10: fatal error: ../metis/libmetis/gklib_defs.h: No such file or directory
parmetis>    23 | #include "../metis/libmetis/gklib_defs.h"
parmetis>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
parmetis> compilation terminated.
parmetis> In file included from /build/source/libparmetis/akwayfm.c:14:
parmetis> /build/source/libparmetis/./parmetislib.h:23:10: fatal error: ../metis/libmetis/gklib_defs.h: No such file or directory
parmetis>    23 | #include "../metis/libmetis/gklib_defs.h"
parmetis>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
parmetis> make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:104: libparmetis/CMakeFiles/parmetis.dir/balancemylink.c.o] Error 1
parmetis> make[2]: *** Waiting for unfinished jobs....
parmetis> compilation terminated.
parmetis> make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:76: libparmetis/CMakeFiles/parmetis.dir/akwayfm.c.o] Error 1
parmetis> In file included from /build/source/libparmetis/ametis.c:15:
parmetis> /build/source/libparmetis/./parmetislib.h:23:10: fatal error: ../metis/libmetis/gklib_defs.h: No such file or directory
parmetis>    23 | #include "../metis/libmetis/gklib_defs.h"
parmetis>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
parmetis> compilation terminated.
parmetis> make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:90: libparmetis/CMakeFiles/parmetis.dir/ametis.c.o] Error 1
parmetis> In file included from /build/source/libparmetis/comm.c:11:
parmetis> /build/source/libparmetis/./parmetislib.h:23:10: fatal error: ../metis/libmetis/gklib_defs.h: No such file or directory
parmetis>    23 | #include "../metis/libmetis/gklib_defs.h"
parmetis>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
parmetis> compilation terminated.
parmetis> make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:118: libparmetis/CMakeFiles/parmetis.dir/comm.c.o] Error 1
parmetis> make[1]: *** [CMakeFiles/Makefile2:184: libparmetis/CMakeFiles/parmetis.dir/all] Error 2
parmetis> make: *** [Makefile:136: all] Error 2
```

I'd prefer to simply use the commits I provided.

Reply:
It’s totally fine to set `enableParallelBuilding = true;`, in my opinion. But it is indeed the overridden `configurePhase` that is the issue here, because CMake’s includes this:

```bash
    if ! [[ -v enableParallelBuilding ]]; then
        enableParallelBuilding=1
        echo "cmake: enabled parallel building"
    fi
```

I think the issue you’re running into there might be because you’re using a relative path whereas the Makefile explicitly makes it absolute, though.